go/vt/vtadmin/cluster/dynamic/interceptors.go

```diff
@@ -0,0 +1,134 @@
+package dynamic
```

Comment:
happy to discuss anything in this file in person if it's helpful; it's .... a lot 😅

Reply:
(actually let me take a stab at _actually_ writing docs first, so it exists for posterity; _then_ i will make my offer)